### ISSUE TITLE
False positive: unused import when using macroexpand

### ISSUE THREAD

**Issue by mrkam2** [clj-kondo/clj-kondo#2246]
**version**

`clj-kondo v2023.12.15`

**problem**

False unused import warnings are generated when type hints are used in the parameters of the macro configured with a `:macroexpand` hook.

`a.clj`:
```clojure
(ns a
  (:import [java.lang Long]))

(defmacro my-defrecord [& body]
  `(defrecord ~@body))

(my-defrecord foo [^Long x])
```

`.clj-kondo/config.edn`:
```edn
{:hooks
{:macroexpand {a/my-defrecord hooks.a/my-defrecord}}
#_{:analyze-call {a/my-defrecord hooks.a/my-defrecord}}
}
```

`.clj-kondo/hooks/a.clj`:
```clojure
(ns hooks.a
  (:require
   [clj-kondo.hooks-api :as api]))

; this doesn't work
(defmacro my-defrecord [& body]
    `(defrecord ~@body))

; this works
#_(defn my-defrecord
  [{:keys [node]}]
  (let [body (rest (:children node))]
    {:node (api/list-node (cons (api/token-node 'defrecord) body))}))
```

Linting fails as following:
```
$ lint a.clj                                                                                                          
a.clj:2:23: warning: Unused import Long
linting took 82ms, errors: 0, warnings: 1
```

**expected behavior**

No unused import warning is generated, as in the case of `:analyze-call` hook provided in the example.

[unused-import-with-macroexpand.zip](https://github.com/clj-kondo/clj-kondo/files/13746898/unused-import-with-macroexpand.zip)

**Comment by borkdude** [comment]
The issue (bug) is that metadata isn't properly preserved in the conversion to s-expressions and back:

``` clojure
(comment
  (def node (clj-kondo.impl.rewrite-clj.parser/parse-string "^Long foo"))
  (:meta node) ;; => :token Long
  (def sexpr (api/sexpr node)) ;; foo
  (meta sexpr)
  (def node2 (api/coerce sexpr))
  (:meta node2) ;;=> nil
  )
```

**Comment by borkdude** [comment]
Compare to the original rewrite-clj:

``` clojure
user=> (require '[rewrite-clj.parser :as p])
nil
user=> (p/parse-string "^Long foo")
<meta: ^Long foo>
user=> (def node (p/parse-string "^Long foo"))
#'user/node
user=> (require '[rewrite-clj.node :as n])
nil
user=> (n/sexpr n)
CompilerException java.lang.RuntimeException: Unable to resolve symbol: n in this context, compiling:(NO_SOURCE_PATH:5:1)
user=> (n/sexpr node)
foo
user=> (meta (n/sexpr node))
{Long true}
user=> (n/coerce (n/sexpr node))
<meta: ^{Long true} foo>
```

**Comment by mrkam2** [comment]
> The issue (bug) is that metadata isn't properly preserved in the conversion to s-expressions and back:

Is this fixable for clj-kondo?
